enum.rb

```diff
@@ -0,0 +1,118 @@
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    index = 0
+    array = to_a
+    while index < array.length
+      yield(array[index])
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_wth_index) unless block_given?
+
+    length.times do |index|
+      yield(self[index], index)
+    end
+    self
+  end
+
+  def my_select
+    return to_enum if block_given? == false
+
+    array = []
+    my_each { |index| array.push(index) if yield(index) }
+    array
+  end
+
+  def my_all?(_arr = nil)
+    condition = true
+    if block_given?
+      my_each { |index| condition = false unless yield(index) }
+    else
+      my_each { |index| condition = false if index.nil? || index == false }
+    end
+    condition
+  end
+
+  def my_any?(arr = nil)
+    condition = false
+    my_each do |index|
+      if block_given?
+        condition = true if yield(index)
+      elsif arr.nil?
+        condition = true unless index
+      elsif arr === index
+        condition = true
+      end
+    end
+    condition
+  end
```

Comment:
- [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

 #### my_any? 

- [ ] when no block or argument is given returns true if at least one of the collection is not false or nil

```ruby     
[nil, false, nil, false].my_any? == [nil, false, nil, false].any? => false
```
- Yours is giving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/79233591-52ba0080-7e26-11ea-9f0c-4f62b5098a65.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/79233705-72e9bf80-7e26-11ea-9bc5-29778c8655dc.png)